File: ZkTony-Manager/app/src/main/kotlin/com/zktony/manager/data/remote/model/Equipment.kt
package com.zktony.manager.data.remote.model

import com.zktony.www.common.extension.currentTime
import java.util.*

/**
 * @author: 刘贺贺
 * @date: 2023-02-16 16:36
 */
data class Equipment(
    val id: String = UUID.randomUUID().toString(),
    val name: String = "",
    val model: String = "",
    val voltage: String = "",
    val power: String = "",
    val frequency: String = "",
    val attachment: String = "",
    val remarks: String = "",
    val create_by: String = "",
    val create_time: String = currentTime()
)

File: ZkTony-Api/database/src/models/manager/equipment.rs
use serde::Deserialize;

#[derive(Deserialize, Debug, Clone)]
pub struct EquipmentSaveReq {
    pub id: String,
    pub name: String,
    pub model: String,
    pub voltage: String,
    pub power: String,
    pub frequency: String,
    pub attachment: String,
    pub remarks: Option<String>,
    pub create_by: Option<String>,
    pub create_time: Option<String>,
}

#[derive(Deserialize, Debug, Clone)]
pub struct EquipmentDeleteReq {
    pub id: String,
}

#[derive(Deserialize, Debug, Clone)]
pub struct EquipmentGetReq {
    pub id: Option<String>,
    pub name: Option<String>,
    pub model: Option<String>,
    pub begin_time: Option<String>,
    pub end_time: Option<String>,
}
